### PR TITLE
luaPackages.luaunit: init at 3.4-1

### PR DESCRIPTION
**Pull request by lockejan**
###### Description of changes

unitttesting for lua

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lockejan** [comment]
propagated build inputs in the generated expression are bringing lua twice.
I guess I have to add something to overrides to mitigate that....

File: pkgs/development/lua-modules/generated-packages.nix [hunk 1]
```diff
@@ -2138,6 +2138,31 @@ buildLuarocksPackage {
...
+    sha256 = "1v8nkiwz2nr242h5cl4af6vmn5gxmn94skps1qhb55ak60j20nvr";
+  };
+
+  disabled = with lua; (luaOlder "5.1") || (luaAtLeast "5.5");
+  propagatedBuildInputs = [ lua lua ];
+
+  meta = {
+    homepage = "http://github.com/bluebird75/luaunit";
+    description = "A unit testing framework for Lua";
+    license.fullName = "BSD";
```
Comment (teto) [inline pkgs/development/lua-modules/generated-packages.nix:2162]: there should be a "maintainer" line here

**Comment by teto** [comment]
> propagated build inputs in the generated expression containing lua twice.

It's fine I think. it will be deduplicater later.

File: pkgs/development/lua-modules/generated-packages.nix (hunk 1 above)
Comment (lockejan) [reply pkgs/development/lua-modules/generated-packages.nix:2163]: done

**Comment by lockejan** [comment]
> > propagated build inputs in the generated expression containing lua twice.
> 
> It's fine I think. it will be deduplicater later.

Hm, ok. Nix refused to evaluate my branch, when leaving it like that.
But I'm not the expert here.

File: pkgs/development/lua-modules/generated-packages.nix [hunk 2]
```diff
@@ -2138,6 +2138,32 @@ buildLuarocksPackage {
...
+    url    = "mirror://luarocks/luaunit-3.4-1.rockspec";
+    sha256 = "111435fa8p2819vcvg76qmknj0wqk01gy9d1nh55c36616xnj54n";
+  }).outPath;
+  src = fetchurl {
+    url    = "https://github.com/bluebird75/luaunit/releases/download/LUAUNIT_V3_4/rock-luaunit-3.4.zip";
+    sha256 = "1v8nkiwz2nr242h5cl4af6vmn5gxmn94skps1qhb55ak60j20nvr";
+  };
+
+  disabled = with lua; (luaOlder "5.1") || (luaAtLeast "5.5");
+  propagatedBuildInputs = [ lua lua ];
```
Comment (SuperSandro2000) [inline pkgs/development/lua-modules/generated-packages.nix:2157]:
```suggestion
  propagatedBuildInputs = [ lua ];
```

**Review by SuperSandro2000 — COMMENTED** [review]
also please fix the commit message

File: pkgs/development/lua-modules/generated-packages.nix (hunk 2 above)
Comment (lockejan) [reply pkgs/development/lua-modules/generated-packages.nix:2157]: I know. This is a generated expression. I wonder if it's possible to fix this via overrides?

**Comment by lockejan** [comment]
> also please fix the commit message

What do you mean? using `at` instead of `->` ?

File: pkgs/development/lua-modules/generated-packages.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/lua-modules/generated-packages.nix:2157]: Maybe the generator needs a fix.

Comment (lockejan) [reply pkgs/development/lua-modules/generated-packages.nix:2157]: Probably, but that's out of scope of this PR I'd say.

**Comment by SuperSandro2000** [comment]
> > also please fix the commit message
> 
> What do you mean? using `at` instead of `->` ?

yes. The format is ``package: init at X`` or ``package: X -> Y``

**Pull request merged**